### PR TITLE
feat(PT-2989): refresh collection meta

### PR DESCRIPTION
**Pull request by dmitry-saraykin**

### CODE REVIEW

File: enrichment/src/main/kotlin/com/rarible/protocol/union/enrichment/model/CollectionMetaRefreshRequest.kt [hunk 1]
```diff
@@ -0,0 +1,18 @@
+package com.rarible.protocol.union.enrichment.model
+
+import org.bson.types.ObjectId
+import org.springframework.data.annotation.Id
+import org.springframework.data.mongodb.core.mapping.Document
+import java.time.Instant
+
+@Document("collection_meta_refresh_request")
```
Comment (yakischik) [inline enrichment/src/main/kotlin/com/rarible/protocol/union/enrichment/model/CollectionMetaRefreshRequest.kt:8]: Lets call it meta_refresh_request

File: worker/src/main/kotlin/com/rarible/protocol/union/worker/config/KafkaConsumerConfiguration.kt [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+        workerProperties: WorkerProperties,
+    ): LagService = LagService(
+        kafkaConsumer = kafkaConsumer,
+        bootstrapServers = producerProperties.brokerReplicaSet,
+        metaConsumerGroup = kafkaGroupFactory.metaDownloadExecutorGroup(KafkaGroupFactory.ITEM_TYPE),
+        refreshTopic = UnionInternalTopicProvider.getItemMetaDownloadTaskExecutorTopic(
+            applicationEnvironmentInfo.name,
+            ItemMetaPipeline.REFRESH.pipeline
+        ),
+        maxLag = workerProperties.collectionMetaRefresh.maxKafkaLag,
```
Comment (yakischik) [inline worker/src/main/kotlin/com/rarible/protocol/union/worker/config/KafkaConsumerConfiguration.kt:43]: We wanted to get rid of it, dont we?

File: worker/src/main/kotlin/com/rarible/protocol/union/worker/job/CollectionMetaRefreshRequestCleanupJob.kt [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+    private val collectionMetaRefreshRequestRepository: CollectionMetaRefreshRequestRepository,
+    properties: WorkerProperties,
+    meterRegistry: MeterRegistry,
+) : SequentialDaemonWorker(
+    meterRegistry = meterRegistry,
+    properties = DaemonWorkerProperties().copy(
+        pollingPeriod = properties.collectionMetaRefreshRequestCleanup.rate,
+        errorDelay = properties.collectionMetaRefreshRequestCleanup.rate
+    ),
+    workerName = "collection-meta-refresh-request-cleanup-job"
```
Comment (yakischik) [inline worker/src/main/kotlin/com/rarible/protocol/union/worker/job/CollectionMetaRefreshRequestCleanupJob.kt:22]: Lets call it like a table "meta_refresh_request_cleanup_job"

File: worker/src/main/kotlin/com/rarible/protocol/union/worker/task/meta/RefreshSimpleHashTask.kt [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+            CollectionMetaRefreshRequest(
+                collectionId = parsedParam.collectionId,
+                full = true,
+            )
+        )
+        logger.info("Finished RefreshSimpleHashTask from=$from, param=$param")
+    }.withTraceId()
+
+    companion object {
+        const val REFRESH_SIMPLEHASH_TASK = "REFRESH_SIMPLEHASH_TASK"
```
Comment (yakischik) [inline worker/src/main/kotlin/com/rarible/protocol/union/worker/task/meta/RefreshSimpleHashTask.kt:42]: Lets call it "META_REFRESH_SIMPLEHASH_TASK"

File: worker/src/test/kotlin/com/rarible/protocol/union/worker/task/meta/RefreshMetaTaskTest.kt [hunk 5]
```diff
@@ -0,0 +1,225 @@
...
+                pipeline = ItemMetaPipeline.REFRESH,
+                force = true
+            )
+        }
+    }
+}
+
+fun createMeta(): DownloadEntry<UnionMeta> = randomItemMetaDownloadEntry()
+
+fun createUnionItem(id: ItemIdDto) = UnionItem(
```
Comment (yakischik) [inline worker/src/test/kotlin/com/rarible/protocol/union/worker/task/meta/RefreshMetaTaskTest.kt:206]: As far as i remember, there were methods in test data factories for these entities. If not, better to create them as all other randomLalala()

File: worker/src/test/resources/application-test.yml [hunk 6]
```diff
@@ -50,6 +50,8 @@ worker:
     enabled: false
   meta-collection-retry:
     enabled: false
+  collection-meta-refresh:
```
Comment (yakischik) [inline worker/src/test/resources/application-test.yml:53]: Lets write it in camelcase, like in Java - its easier to search such properties in project when naming the same

File: worker/src/main/kotlin/com/rarible/protocol/union/worker/job/meta/RefreshMetaTaskSchedulingJob.kt [hunk 7]
```diff
@@ -0,0 +1,116 @@
...
+    properties: WorkerProperties,
+    meterRegistry: MeterRegistry,
+    private val handler: RefreshMetaTaskSchedulingJobHandler,
+) : SequentialDaemonWorker(
+    meterRegistry = meterRegistry,
+    properties = DaemonWorkerProperties().copy(
+        pollingPeriod = properties.collectionMetaRefresh.rate,
+        errorDelay = properties.collectionMetaRefresh.rate,
+    ),
+    workerName = "item-meta-refresh-task-scheduling-job"
```
Comment (yakischik) [inline worker/src/main/kotlin/com/rarible/protocol/union/worker/job/meta/RefreshMetaTaskSchedulingJob.kt:48]: meta_refresh_request_job

**Review by yakischik — APPROVED** [review]

File: api/src/main/kotlin/com/rarible/protocol/union/api/controller/internal/MetaRefreshController.kt [hunk 8]
```diff
@@ -0,0 +1,119 @@
...
+@RestController
+class MetaRefreshController(
+    private val collectionMetaRefreshRequestRepository: CollectionMetaRefreshRequestRepository,
+    private val unionMetaProperties: UnionMetaProperties,
+    private val taskRepository: TaskRepository,
+    private val objectMapper: ObjectMapper,
+) {
+
+    @PostMapping(
+        value = ["/maintenance/meta/collections/refresh/full"],
```
Comment (yakischik) [inline api/src/main/kotlin/com/rarible/protocol/union/api/controller/internal/MetaRefreshController.kt:31]: /maintenance/items/meta/refresh
queryParam mode=FULL/ONLY_EMPTY

File: api/src/main/kotlin/com/rarible/protocol/union/api/controller/internal/MetaRefreshController.kt [hunk 9]
```diff
@@ -0,0 +1,119 @@
...
+    ): Unit = withTraceId {
+        scheduleRefresh(
+            collections = parseCollectionsFromBody(body),
+            full = false,
+            scheduledAt = scheduledAt ?: nowMillis()
+        )
+    }
+
+    @DeleteMapping(
+        value = ["/maintenance/meta/collections/refresh/cancel"],
```
Comment (yakischik) [inline api/src/main/kotlin/com/rarible/protocol/union/api/controller/internal/MetaRefreshController.kt:63]: /maintenance/items/meta/refresh DELETE

File: api/src/main/kotlin/com/rarible/protocol/union/api/controller/internal/MetaRefreshController.kt [hunk 10]
```diff
@@ -0,0 +1,119 @@
...
+        produces = ["application/json"]
+    )
+    suspend fun cancelRefresh(): Unit = withTraceId {
+        logger.info("Cancelling refresh")
+        collectionMetaRefreshRequestRepository.deleteAll()
+    }
+
+    @GetMapping(value = ["/maintenance/meta/collections/refresh/status"], produces = ["text/plain"])
```
Comment (yakischik) [inline api/src/main/kotlin/com/rarible/protocol/union/api/controller/internal/MetaRefreshController.kt:71]: /maintenance/items/meta/refresh GET

File: api/src/main/kotlin/com/rarible/protocol/union/api/controller/internal/MetaRefreshController.kt [hunk 11]
```diff
@@ -0,0 +1,119 @@
...
+        produces = ["application/json"]
+    )
+    suspend fun emptyOnlyRefresh(
+        @RequestParam(required = false) scheduledAt: Instant?,
+        @RequestBody body: String
+    ): Unit = withTraceId {
+        scheduleRefresh(
+            collections = parseCollectionsFromBody(body),
+            full = false,
+            scheduledAt = scheduledAt ?: nowMillis()
```
Comment (yakischik) [inline api/src/main/kotlin/com/rarible/protocol/union/api/controller/internal/MetaRefreshController.kt:58]: withSimpleHash

File: worker/src/main/kotlin/com/rarible/protocol/union/worker/config/KafkaConsumerConfiguration.kt (hunk 2 above)
Comment (dmitry-saraykin) [reply worker/src/main/kotlin/com/rarible/protocol/union/worker/config/KafkaConsumerConfiguration.kt:43]: yes but in another task. this one is too big already

**Pull request merged**